Docs/source/Software/Tasks.rst

```diff
@@ -187,6 +187,38 @@ Additional Considerations
     All possible CAN messages that will be sent to the rest of the car's system by the BPS
     are listed on the `CAN Bus IDs spreadsheet <https://docs.google.com/spreadsheets/d/11YWoMVZw8BFr8kyO4DIz0g-aIU_vVa0d-WioSRq85TI/edit#gid=0>`_.
 
+Pet WatchDog Task: Harshita Gorla & Clark Poon
+==============================================
+
+Purpose
+    The purpose of this task is to recognize if the BPS RTOS has stalled. If it is stuck somewhere
+    in the code, the car must shut down.
+
+Functionality
+    This task checks the ``WDog_BitMap`` variable to see if the 3 LSB are set by the temperature,
+    voltage, balancing, and current tasks. If these tasks ran and set those bits, that means that the BPS
+    is functional and the timer is reset. If those bits are not set, the timer will keep running
+    and a fault handler will be called.
+
+Priority
+    This task is priority 3. It is above the voltage, temperature, balancing, and current monitoring tasks
+    because if they run before the WatchDog timer is reset, the BPS will fault even if it is 
+    working correctly.
```

Comment:
More specifically, we do not want this thread to be blocked by the other periodic threads, since the watchdog timer will reset the BPS if this thread is not called. TBH I am not sure how much it matters that this thread has a higher priority than the threads it is monitoring, since they will also cause the watchdog timer to reset the car if they are blocked for too long

Reply:
I opened #312 for this